hummingbot/strategy/pure_market_making/inventory_skew_single_size_sizing_delegate.pyx

```diff
@@ -0,0 +1,135 @@
+from hummingbot.market.market_base import MarketBase
+from hummingbot.core.event.events import (
+    OrderType,
+    TradeType
+)
+from hummingbot.logger import HummingbotLogger
+from .data_types import SizingProposal
+from .pure_market_making_v2 cimport PureMarketMakingStrategyV2
+
+s_logger = None
+s_decimal_0 = Decimal(0)
+
+
+cdef class InventorySkewSingleSizeSizingDelegate(OrderSizingDelegate):
+
+    def __init__(self, order_size: float, inventory_target_base_percent: Optional[float] = None):
+        super().__init__()
+        self._order_size = order_size
+        self._inventory_target_base_percent = inventory_target_base_percent
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        global s_logger
+        if s_logger is None:
+            s_logger = logging.getLogger(__name__)
+        return s_logger
+
+    @property
+    def order_size(self) -> float:
+        return self._order_size
+
+    cdef object c_get_order_size_proposal(self,
+                                          PureMarketMakingStrategyV2 strategy,
+                                          object market_info,
+                                          list active_orders,
+                                          object pricing_proposal):
+        cdef:
+            MarketBase market = market_info.market
+            str trading_pair = market_info.trading_pair
+            object buy_fees
+            object base_asset_balance = Decimal(market.c_get_available_balance(market_info.base_asset))
+            object quote_asset_balance = Decimal(market.c_get_available_balance(market_info.quote_asset))
+            object top_bid_price
+            object top_ask_price
+            object mid_price
+            object total_base_asset_quote_value
+            object total_quote_asset_quote_value
+            object current_base_percent
+            object current_quote_percent
+            object target_base_percent
+            object target_quote_percent
+            object current_target_base_ratio
+            object current_target_quote_ratio
+            object bid_order_size = Decimal(self._order_size)
+            object ask_order_size = Decimal(self._order_size)
+            object quantized_bid_order_size
+            object quantized_ask_order_size
+            bint has_active_bid = False
+            bint has_active_ask = False
+            object required_quote_asset_balance
+
+        for active_order in active_orders:
+            if active_order.is_buy:
+                has_active_bid = True
+                quote_asset_balance += active_order.quantity * active_order.price
+            else:
+                has_active_ask = True
+                base_asset_balance += active_order.quantity
+
+        if has_active_bid and has_active_ask:
+            return SizingProposal([s_decimal_0], [s_decimal_0])
+
+        if self._inventory_target_base_percent is not None:
+            top_bid_price = Decimal(market.c_get_price(trading_pair, False))
```

Comment:
The scaling logic is pretty hard to read. I'd recommend adding at least two in-code documentation.

 1. The overall intention of this.
 2. Some rough explanation about the math. (esp. the `2 - ...` thing)